apollo-federation/src/sources/connect/json_selection/lit_expr.rs

```diff
@@ -317,6 +325,24 @@ mod tests {
             "-123.",
             LitExpr::Number(serde_json::Number::from_f64(-123.0).unwrap()),
         );
+        check_parse("00", LitExpr::Number(serde_json::Number::from(0)));
+        check_parse(
+            "-00",
+            LitExpr::Number(serde_json::Number::from_f64(-0.0).unwrap()),
+        );
+        check_parse("0", LitExpr::Number(serde_json::Number::from(0)));
+        check_parse(
+            "-0",
+            LitExpr::Number(serde_json::Number::from_f64(-0.0).unwrap()),
+        );
+        check_parse(" 00 ", LitExpr::Number(serde_json::Number::from(0)));
+        check_parse(" 0 ", LitExpr::Number(serde_json::Number::from(0)));
+        check_parse(
+            " - 0 ",
+            LitExpr::Number(serde_json::Number::from_f64(-0.0).unwrap()),
+        );
+        check_parse("001", LitExpr::Number(serde_json::Number::from(1)));
+        check_parse("0010", LitExpr::Number(serde_json::Number::from(10)));
```

Comment:
Just in case, could we add a test case for leading zeros on fractional numbers? Like `-007.` (actually generated in one of the fuzz cases) or `000.10`?

Reply:
https://github.com/apollographql/router/pull/6766/commits/bfccf2a3a0f8fa7ebfbc495a4c8e616ddb3443c5